cmd/kubelet/app/server.go

```diff
@@ -361,15 +353,37 @@ func (s *KubeletServer) createAPIServerClient() (*client.Client, error) {
 	if len(s.APIServerList) > 1 {
 		glog.Infof("Multiple api servers specified.  Picking first one")
 	}
-	clientConfig.Host = s.APIServerList[0]
 
-	s.addChaosToClientConfig(&clientConfig)
+	// This creates a client, first loading any specified kubeconfig
+	// file, and then overriding the Master flag, if non-empty.
+	clientConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
Setting --auth-path (which is deprecated) will only be overridden by --kubeconfig if you happen to have a valid kubeconfig file at the default location. This seemed unlikely enough (and we want to rip out auth path support soonish) that I wasn't convinced it was worth jumping through extra hoops here.

A quick look into the options you proposed shows that util's StringFlag is currently only used in kubectl (I'm not sure why) and that there isn't a Changed property in [pflag.Flag](https://github.com/ogier/pflag/blob/master/flag.go#L142).